bundles/org.openhab.binding.enturno/src/main/java/org/openhab/binding/enturno/internal/connection/EnturNoConnection.java

```diff
@@ -235,11 +232,10 @@ private DisplayData getDisplayData(StopPlace stopPlace, Map<String, List<Estimat
     }
 
     private String getIsoDateTime(String dateTimeWithoutColonInZone) {
-        String dateTime = StringUtils.substringBeforeLast(dateTimeWithoutColonInZone, "+");
-        String offset = StringUtils.substringAfterLast(dateTimeWithoutColonInZone, "+");
+        String dateTime = dateTimeWithoutColonInZone.substring(0, dateTimeWithoutColonInZone.lastIndexOf("+"));
```

Comment:
This assumes a well-formatted input string with time-zone.

If "+" is not contained in the string, `lastIndexOf` returns -1, and you will call `substring(0, -1)` which will then throw `IndexOutOfBoundsException`. `substringBeforeLast` would return the input string itself in this case.

`substringAfterLast` returns empty string if not found, and I believe `StringUtils.substring` is also quite forgiving, so I believe the end result for an input string without time-zone would be okay - at least no exceptions would be thrown.

Reply:
Added multiple checks for `IndexOutOfBoundsException` and made all alternatives to the apache lib mimic them mroe closely.